git/intermediate_git/a_deeper_look_at_git.md

```diff
@@ -172,59 +172,75 @@ Let's review the dangers we've addressed so far. I know, I know, it's scary stuf
+So what is a branch? Based off of your exposure, you might be visualizing a branch as a group of commits. This actually isn't the case! **A branch is actually a pointer to a single commit!** Hearing this, your first thought might be _"Well if a branch is just a finger pointing at a single commit, how does that single commit know about all the commits that came before it?"_ The answer to this question is very simple: Each commit is also a pointer that points to the commit that came before it! Wow. This might be a lot to take in, so let's take a moment to absorb that fact.
 
 Now that you've had a second to gather your thoughts and attempt to wrap your head around this concept, it might help to go back and look at a concrete example of pointers we used in this lesson. Let's think back to our use of `git rebase -i HEAD~2`. If you can remember, this command lets us edit the last two commits. Do you have any guesses on how Git knew which two commits to edit? That's right, by using pointers! We start at HEAD, which is a special pointer for keeping track of the branch you're currently on. HEAD points to our most recent commit in the current branch. That commit points to the commit made directly before it, which we can call commit two. That's how `git rebase -i HEAD~2` starts with a HEAD pointer, and then follows subsequent pointers to find which two commits to edit.
 
 You might be feeling overwhelmed at this point, so let's recap what we've learned. A branch is simply a pointer to a single commit. A commit is a snapshot, and it's a pointer to the commit directly behind it in history. That's it!
 
+### Looking into your Git History
+
+You know how it is often said that you need to learn from the past in order to move into the future? Well, the same applies when it comes to your Git history.
+At this point, you probably know to view your commits by using the `git log` command or by checking your repository on GitHub. While that's effective, here's a stronger and simpler way to see your commits. The `git log --oneline`command. With this command, the terminal displays each commit on a single line.
+
+Now you may be wondering _what_ can we do with this information? Remember the command `git checkout`? This has another use-case when viewing your commit history. In the command line, you can use `git checkout <commit ID>` to view commits in another branch. This is useful if you are working with another developer on a project and want to see their code.
```

Comment:
I find this paragraph confusing. If I was working with another developer and wanted to see their branch, I would check out their branch, not just a commit on their branch.

I believe a better use case would be if I broke some functionality and I wanted to go back a commit or two to see how it was working before.

I do not believe this paragraph is thorough enough, because when checking out a commit like this, it will put them in a 'detached HEAD' state, which can cause confusion.